src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -180,7 +180,7 @@ public function searchBarAction(Request $request)
             $this->cspHandler->disableCsp();
         }
 
-        if (null === $session = $request->getSession()) {
+        if (!$request->hasSession() || null === ($session = $request->getSession())) {
```

Comment:
I'd suggest to use `if (!$request->hasSession() || !$session = $request->getSession()) {`